extensions/sql-database-projects/src/dialogs/createProjectFromDatabaseDialog.ts

```diff
@@ -424,4 +466,71 @@ export class CreateProjectFromDatabaseDialog {
 			throw new Error(constants.extractTargetRequired);
 		}
 	}
+
+	async validate(): Promise<boolean> {
+		try {
+			// the selected location should be an existing directory
+			const parentDirectoryExists = await exists(this.projectLocationTextBox!.value!);
+			if (!parentDirectoryExists) {
+				this.showErrorMessage(constants.ProjectParentDirectoryNotExistError(this.projectLocationTextBox!.value!));
```

Comment:
why not just throw errors inside the try block and let the catch handle the display and return.  Simplifies the code path and consolidates handling logic.

Reply:
Better yet, just catch inside the code that calls `validate()`.  May as well let the original error bubble up as far as it can until we need to transform that into an error banner or something.  Makes this function more reusable.

Reply:
good point. This also happens in the data workspace dialogs, so I'll update those in a separate PR since the tests there would need to be updated too.